### ISSUE TITLE
Unable to select date in calendar - eventListener on auro-input not firing

### ISSUE THREAD

**Issue by shonore-aa** [AlaskaAirlines/auro-datepicker#93]
## Describe the bug
Users are not able to select a date from the auro-datepicker's calendar

## To Reproduce

**This issue only occurs occasional. I have not been able to intentionally recreate this issue**
1. Go to 'https://www.alaskaair.com/search'
2. Click on the datepicker field in the planbook
3. Try and select a date from the datepicker

## Expected behavior

The user should be able to select a date from the datepicker calendar

## Screenshots
https://alaskaair-my.sharepoint.com/:v:/p/stephanie_honore1/ESU4e4Pcl5BMm2gXfKvrZCcBlgOl3qTYbWTG93YpVsKVjQ?e=mvhCRO

## Desktop (please complete the following information):

 - OS: Mac OSX
 - Browser: Chrome
 - Version: latest

## Additional context

This issue occurs sporadically and inconsistently. I am not able to recreate this issue intentionally. From debugging in the browser it could be an auro-input issue.

**Comment by blackfalcon** [comment]
We do not have access to the screen capture link you provided. Also, unless there is a consistent way to repro this issue, that makes it very difficult to troubleshoot. 

@shonore-aa are you able to repro this issue at all in the Auro site? https://auro.alaskaair.com/components/auro/datepicker/api

**Comment by shonore-aa** [comment]
> We do not have access to the screen capture link you provided. Also, unless there is a consistent way to repro this issue, that makes it very difficult to troubleshoot.
> 
> @shonore-aa are you able to repro this issue at all in the Auro site? https://auro.alaskaair.com/components/auro/datepicker/api

@blackfalcon I updated the link so you should have permission to view it. I have not been able to reproduce on the auro site, but I can spend more time trying to test it again on the auro site. It only happens sporadically in the flight search microsite. For some reason the event listener is not firing on the auro-datepicker ... I am curious if this is because it is not ready before the request is made?

**Comment by shonore-aa** [comment]
@blackfalcon I have not been able to reproduce on the auro site

**Comment by shonore-aa** [comment]
Working with @braven112 on this

**So I have a hypothesis:** 

TLDR; 
Basically I think there is a race condition occurring due to JS having a single thread of execution that is shared with the page rendering so the DOM is not getting updated. 

Erika noticed that the issue occurred after she had a tab open with the datepicker for a while. I notice there is a checkRediness() function that checks if the the auro dropdown, auro input and auro calendar are ready. If they are not ready then the function is recursively called to check if all 3 are ready again. 
On starting line 530 I see the following 
 
{

      // Start a retry counter to limit the retry count

      if (!this.readyRetryCount && this.readyRetryCount !== 0) {

        this.readyRetryCount = 0;

      } else {

        this.readyRetryCount += 1;

      }



      const readyTimer = 0;

      const readyRetryLimit = 200;



      if (this.readyRetryCount <= readyRetryLimit) {

        setTimeout(() => {

          this.checkReadiness();

        }, readyTimer);

      }

    }

**Comment by shonore-aa** [comment]
I was able to recreate the issue on https://auro.alaskaair.com/components/auro/datepicker. I had the window sitting for 4.5 hours

https://user-images.githubusercontent.com/107429014/207180180-bc1daa7e-7c9d-46e6-ae9c-24e69e8bc4a1.mov

**Comment by braven112** [comment]
Just to confirm with everyone here what we know about this so far...please add any details I've missed.

Sounds like the working theory at the moment is the the browser is somehow causing, at least some, scripts to stop at running after a long period of inactivity.  This could effect more than just auro-datepicker but that hasn't been explored as of yet.

Short term fixes could include some sort of page level timeout that requires the page to reload after some period of inactivity.  I believe we already do that on legacy shopping pages after 20 minutes, but I'm not certain.  

Long term fixes aren't known but we are asking the principal community for help to see if they have any insights.  Its possible a refactor could help address it but we aren't sure that it would and its not a trivial amount of work to try.  Hoping to gather more information to understand the issue better.

Do we know if this happens in all browsers or just chrome?  Are there other components that this may also effect that we should test?

**Comment by blackfalcon** [comment]
Feedback from another thread. 

![Uploading Screen Shot 2022-12-20 at 8.19.55 AM.png…]()
